### ISSUE TITLE
Entity Label Lookup & Entity URI Lookup not working in Faceted browser

### ISSUE THREAD

**Issue by rapw3k** [openlink/virtuoso-opensource#671]
Hi,

I loaded millions of triples in a virtuoso instance, which was already running for some time (with faceted browser installed).

After the massive population, I noticed that virtuoso checkpoints were not carried out automatically (every 60 min as in `virtuoso.ini`), so the transaction file was growing very large (so I was making checkpoints manually).

The fct interface was working but not all the triples were exposed.

So, I followed the [VOS guidelines for the post installation of fct](http://vos.openlinksw.com/owiki/wiki/VOS/VirtFacetBrowserInstallConfig#Post%20Installation).
 
However, when I tried second step, `VT_INC_INDEX_DB_DBA_RDF_OBJ ();`, the operation seemed to hang; I waited almost 12 before stoping it. 

So, I executed `DB.DBA.RDF_OBJ_FT_RECOVER();` as suggested in https://github.com/dkmfbk/knowledgestore/wiki/Virtuoso-full-text-indexing 

It took several hours but finished. 

I executed again the second step, and after few hours it finished. I executed then `urilbl_ac_init_db()` and finally `s_rank()`, which took quite some time but eventually finished.

Now everything seems to be working fine, i.e., checkpoint are executed automatically, and the fct has indexed all the triples and I can navigate all the store without problems starting from "text search". However, the Entity Label Lookup & Entity URI Lookup are not working.

Could anybody advise me what should I check, or how could I try to fix it?

The endpoint is http://www.foodie-cloud.org/fct/ (which is the official entry point to the real endpoint: http://foodie-vm3.man.poznan.pl/fct/). Try e.g., sevilla 

As reference please see also below the output of status();

```
SQL> status();
REPORT
VARCHAR
_______________________________________________________________________________

OpenLink Virtuoso  Server
Version 07.20.3214-pthreads for Linux as of Dec  4 2015 
Started on: 2017-08-08 23:43 GMT+2
 
Database Status:
  File size 0, 19833600 pages, 8640419 free.
  680000 buffers, 645589 used, 3 dirty 0 wired down, repl age 1440165 0 w. io 0 w/crsr.
  Disk Usage: 301464668 reads avg 0 msec, 0% r 0% w last  1645 s, 323782816 writes flush      31.06 MB,
    890538 read ahead, batch = 104.  Autocompact 33860771 in 27052658 out, 20% saved col ac: 650884 in 14% saved.
Gate:  7859259 2nd in reads, 0 gate write waits, 0 in while read 0 busy scrap. 
Log = /home/virtuoso/virtuoso-7.2.1/var/lib/virtuoso/db/virtuoso.trx, 1628 bytes
11190450 pages have been changed since last backup (in checkpoint state)
Current backup timestamp: 0x0000-0x00-0x00
Last backup date: unknown
Clients: 99 connects, max 9 concurrent
RPC: 7307 calls, -1186 pending, 1 max until now, 0 queued, 568 burst reads (7%), 3 second 0M large, 756M max
Checkpoint Remap 2000 pages, 0 mapped back. 6878 s atomic time.
    DB master 19833600 total 8640419 free 2000 remap 1 mapped back
   temp  89856 total 89851 free
 
Lock Status: 0 deadlocks of which 0 2r1w, 558760 waits,
   Currently 1 threads running 0 threads waiting 0 threads in vdb.
```

**Comment by HughWilliams** [comment]
I would suggest upgrading to the latest develop/7  3217 build (https://github.com/openlink/virtuoso-opensource) ...  as the 3214 build from 2015 is rather old ...

**Comment by rapw3k** [comment]
Thanks @HughWilliams,  all,  I upgraded to the latest build 3217, and still Entity Label Lookup & Entity URI Lookup is not working. 

I also executed the command `urilbl_ac_init_db()` which [says that is the procedure to periodically populate label lookup tables and activate the Label text box of the Entity Label Lookup tab](https://virtuoso.openlinksw.com/dataspace/doc/dav/wiki/Main/VirtFacetBrowserInstallConfig).

Still not working.  

Using the public URL of the endpoint, the result is immediate: no answer (please try it yourself using the URL below). Using the private URL of the endpoint, the loading icon keeps rolling indefinitely, but no answers (see picture attached).

Public FCT endpoint:  http://www.foodie-cloud.org/fct/ 

Can you give me some support?
thanks,
Raul

<img width="1036" alt="screen shot 2018-03-30 at 14 33 27" src="https://user-images.githubusercontent.com/557493/38137932-1a336264-3428-11e8-8950-a9450d207b9b.png">

As reference:
```
SQL> status();
REPORT
VARCHAR
_______________________________________________________________________________

OpenLink Virtuoso  Server
Version 07.20.3217-pthreads for Linux as of Mar 27 2018 
Started on: 2018-03-27 15:01 GMT+2
 
Database Status:
  File size 0, 20208128 pages, 6293397 free.
  1360000 buffers, 1348638 used, 5 dirty 0 wired down, repl age 1033103 0 w. io 0 w/crsr.
  Disk Usage: 29500042 reads avg 0 msec, 0% r 0% w last  82588 s, 2146416 writes flush      55.38 MB,
    33827 read ahead, batch = 41.  Autocompact 190251 in 161657 out, 15% saved col ac: 84573 in 7% saved.
Gate:  19935 2nd in reads, 0 gate write waits, 0 in while read 0 busy scrap. 
Log = /home/virtuoso/virtuoso/var/lib/virtuoso/db/virtuoso.trx, 1877 bytes
13911923 pages have been changed since last backup (in checkpoint state)
Current backup timestamp: 0x0000-0x00-0x00
Last backup date: unknown
Clients: 54 connects, max 6 concurrent
RPC: 8277 calls, -889 pending, 1 max until now, 0 queued, 51 burst reads (0%), 0 second 6M large, 408M max
Checkpoint Remap 2000 pages, 0 mapped back. 703 s atomic time.
    DB master 20208128 total 6293397 free 2000 remap 1 mapped back
   temp  1536 total 1531 free
```

**Comment by HughWilliams** [comment]
You appear to have updated the Virtuoso binary to the latest `develop/7` build, but not the Faceted Browser and any other VADs to the latest `develop/7` builds.

As you current FCT version is:
```
Faceted Search & Find service v1.13.68 as of Nov 06 2015
```

where as the version from my last build is:
```
Faceted Search & Find service v1.13.91 as of Feb 06 2018
```

Thus please build and upgrade to the latest version ....

**Comment by rapw3k** [comment]
Dear @HughWilliams , all, this might be a basic question, but where can I find the latest release or source of the FCT ? The only place I find is https://virtuoso.openlinksw.com/download/ . Is this the correct version? I cannot see which version is it...

Note that for the installation of the latest version of virtuoso, I downloaded the latest release from https://github.com/openlink/virtuoso-opensource/releases 
Below a screenshot from my conductor.


thanks...

<img width="1114" alt="screen shot 2018-04-03 at 14 11 10" src="https://user-images.githubusercontent.com/557493/38248553-09d41ca0-3749-11e8-947f-2d50ec20f76a.png">

**Comment by rapw3k** [comment]
hi again,
I tried with two different versions I was able to find (version 1.13.91 couldnt find it anywhere.):
* version 1.15_git65 available from https://virtuoso.openlinksw.com/download/ or http://opldownload.s3.amazonaws.com/uda/vad-packages/7.2/fct_dav.vad 
* version 1.13.88 available from https://virtuoso.openlinksw.com/dataspace/doc/dav/wiki/Main/VOSDownload 


First I upgraded to version 1.15_git65, but it was worse. I received the error below, when clicking on a resource (describe):
Error 42001
SR185: Undefined procedure DB.DBA.http_sparql_object.
Then I removed the fct and installed version 1.13.88. Now works as before. Text search works fine, but the Entity Label and Entity URI lookup are not working. 

So, in summary, the new version didnt change anything.
Any help/suggestions/pointers?
thanks in advance,
Raul

**Comment by rapw3k** [comment]
This issue has been solved... It was not related to the fct but with a proxy configuration. There are two related issues/questions, but maybe they require different thread?

1. Where to find fct version 1.13.91?

2. There is an issue with fct version 1.15_git65 when opening (describe) a resource: 
```
Error 42001
SR185: Undefined procedure DB.DBA.http_sparql_object.
```

**Comment by TallTed** [comment]
Hello, @rapw3k --

For VOS, the latest pre-built VAD packages are available via <http://vos.openlinksw.com/owiki/wiki/VOS/VOSDownload> or <http://download3.openlinksw.com/index.html?prefix=uda/vad-vos-packages/>, and the latest pre-built FCT for VOS 7.2 is <http://download3.openlinksw.com/uda/vad-vos-packages/7.2/fct_dav.vad>.

(Note -- the latest pre-built VAD is older than you will get by building yourself from the `/develop/7/` branch.)

The pre-built VAD packages available from <https://virtuoso.openlinksw.com/download/> or <http://download3.openlinksw.com/index.html?prefix=uda/vad-packages/> are for the Enterprise Edition of Virtuoso, not the Open Source Edition.

**Comment by rapw3k** [comment]
Thanks for the clarification @TallTed . 
Now I am wondering how to build myself the VAD package. 
As a test, I tried building virtuoso from develop7 branch in my mac, but it only generates conductor_dav.vad and isparql_dav.vad. How to build the fct package?

**Comment by TallTed** [comment]
@rapw3k - You should just need to add `--enable-fct-vad` (or `--enable-all-vads`) to your `./configure ...`

**Comment by rapw3k** [comment]
Great, thanks @TallTed, didn't know about those options.

**Comment by TallTed** [comment]
For future reference, `./configure --help` reveals lots of potentially interesting info...
